docs/usage/automerge-configuration.md

```diff
@@ -0,0 +1,115 @@
+And this is especially so if your repository needs rebasing, e.g. because you use lock files. e.g. let's say you have dependencies `abc` and `xyz` with upgrades, and you use a `yarn.lock` file.
+
+- At the start of the schedule, `Renovate` will create branches for `abc` and `xyz` upgrades, including `yarn.lock` updates
+- After `abc` passes tests, `Renovate` will automerge it to `master`
+- The `xyz` branch probably now has `yarn.lock` conflicts
+- Renovate will immediately check all other branches and rebase them
+- The change to `xyz` branch will trigger another round of CI tests
+- After the updated `xyz` branch passes, Renovate will automerge it too
+
+This is a lot better than you waking up to two PRs and then having to deal with conflicts yourself after you merge the first one.
+
+Remember our running `eslint` example? Let's automerge it if all the linting updates pass:
+
+```json
+  "packageRules": [
+    {
+      "packagePatterns": [ "eslint" ],
+      "groupName": "eslint",
+      "schedule": ["before 2am on monday"],
+      "automerge": true,
+      "automergeType": "branch"
+    }
+  ]
+```
+
+Have you come up with a rule that you think others would benefit from?
+How about a PR back to [renovate-config](https://github.com/singapore/renovate-config) with the above rule named `":automergeEslintWeekly"` ?
+
+## Frequent problems and how to resolve them
+
+### Automerge config not correct to begin with
+
+What are the most common mistakes people make when doing the initial configuring?
+
+### Branch vs PR
+
+Those of you familiar with GitHub might note that even if you automerge PRs, you are still going to get notifications (noise) anyway - one when the PR is created and another when it is merged.
+For this reason we recommend you consider setting `automergeType=branch` which will mean:
+
+- Renovate first creates a branch and no PR
+- If tests pass, Renovate pushes a commit directly to `master` without PR
+- If tests fail, Renovate raises a PR for you to review
+
+The result is that passing updates are essentially "silent" - the only sign of them are the commits to your `master` branch.
+
+### Absence of tests
+
+If you have no tests but still want Renovate to automerge, you need to add "requiredStatusChecks": null to your configuration.
+
+### Branch protection rules
+
+What branch protection rules specifically are causing problems?
+
+### Required reviews
+
+If you have turned on "Require pull request reviews before merging" in your GitHub branch protection settings, you'll need a way to approve the automerge Renovate pull requests.
+The Renovate team has made two apps to automate the approval.
+
+If you require one approval it's enough to install and use the [renovate-approve](https://github.com/apps/renovate-approve) app.
+But, if you require two approvals, then you also need to install and use the [Renovate Approve 2](https://github.com/apps/renovate-approve-2) app.
```

Comment:
The `renovate-approve` and `renovate-approve-2` apps do not have a similar user-facing name on the "GitHub app store". It's also a bit weird to link to the appstore, as GitHub themselves seem to only use the Marketplace?
Is this all intended behavior?

Reply:
There's no requirement to add to marketplace. Previously it was only for paid apps anyway